fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPreprocessor.java

```diff
@@ -1874,7 +1937,18 @@ public void computeScanRangeAssignment() throws Exception {
                 if (minLocation == null) {
                     throw new UserException("Scan range not found" + backendInfosString(false));
                 }
-                assignedBytesPerHost.put(minLocation.server, assignedBytesPerHost.get(minLocation.server) + 1);
+
+                // only enable for load now, The insert into select performance problem caused by data skew is the most serious
+                if (Config.enable_schedule_insert_query_by_row_count && isLoad
+                        && scanRangeLocations.getScan_range().isSetInternal_scan_range()
+                        && scanRangeLocations.getScan_range().getInternal_scan_range().isSetRow_count()
+                        && scanRangeLocations.getScan_range().getInternal_scan_range().getRow_count() > 0) {
```

Comment:
How about also abstracting the condition to a variable, since it is too long?
```suggestion
                boolean isAssignedByRowCount = Config.enable_schedule_insert_query_by_row_count && isLoad
                        && scanRangeLocations.getScan_range().isSetInternal_scan_range()
                        && scanRangeLocations.getScan_range().getInternal_scan_range().isSetRow_count()
                        && scanRangeLocations.getScan_range().getInternal_scan_range().getRow_count() > 0;
                if (isAssignedByRowCount) {
```